broker-core/src/main/java/org/apache/qpid/server/model/adapter/FileBasedGroupProviderImpl.java

```diff
@@ -205,17 +205,22 @@ public String getPath()
             {
                 throw new IllegalConfigurationException(String.format("Group provider '%s' is not activated. Cannot create a group.", getName()));
             }
+            Set<String> availableGroups = _groupDatabase.getAllGroups();
+            if(!availableGroups.contains(groupName))
```

Comment:
If a group provider is configured with `caseSensitive==false`, the group lookup should be case insensitive. I would like to suggest to move group existence check into `Database#createGroup(String)` as database implementation already has a code to perform case insensitive group lookup. If group exists, an implementation of `Database#createGroup(String)` can throw `IllegalConfigurationException`

Reply:
Done!